src/irmin-pack/unix/file_manager.ml

```diff
@@ -398,16 +401,29 @@ struct
     let pl1 : Payload.t = Control.payload t.control in
     if pl0 = pl1 then Ok ()
     else
-      (* Step 3. Reopen files if generation changed. *)
+      (* Step 3. Reopen files if generation or chunk_num changed. *)
       let* () =
         let gen0 = generation pl0.status in
         let gen1 = generation pl1.status in
+        let chunk_num0 = pl0.chunk_num in
+        let chunk_num1 = pl1.chunk_num in
+        let chunk_start_idx0 = pl0.chunk_start_idx in
+        let chunk_start_idx1 = pl1.chunk_start_idx in
+        let* () =
+          if
+            chunk_num0 <> chunk_num1
+            || chunk_start_idx0 <> chunk_start_idx1
+               (* TODO: remove this last check, kept for passing the tests. *)
+            || gen0 <> gen1
```

Comment:
Why should this be removed? The generation may change independently from the chunks

Reply:
Ah never mind. this branch doesn't modify the suffix/mappin